src/software/new_geom/util/closest_point_test.cpp

```diff
@@ -46,3 +48,43 @@ TEST(ClosestPointTest, point_far_from_line)
     EXPECT_EQ(closestPointOnLine(p, l), expected);
     EXPECT_EQ(closestPointOnLine(l, p), expected);
 }
+
+TEST(GeomUtilTest, test_closest_lineseg_intermediate_point)
```

Comment:
can we add a test for a degenerate segment?

Reply:
Added.